Db 226459 Update HikariCP to 5.1.0 version

asmoiseenko opened the pull request.

sullis committed:
add 'Google Spanner' JDBC to README

brettwooldridge committed:
[maven-release-plugin] prepare for next development iteration

brettwooldridge committed:
Update README.md

sullis committed:
mockito 3.7.7 (#1722)

Co-authored-by: Brett Wooldridge <brett.wooldridge@gmail.com>

darkslave86 committed:
Update HikariConfig.java (#1761)

fix alignment for config lines in log

brettwooldridge committed:
Shorten test (+3 squashed commits)
Squashed commits:
[3b09c167] Javadoc plugin version
[13079b14] Javadoc plugin version
[17a3eb97] Javadoc plugin version

brettwooldridge committed:
Convert to Java 11

brettwooldridge committed:
logging pattern in test (+1 squashed commit)
Squashed commits:
[db19b885]  plugin version

brettwooldridge committed:
Update .travis.yml

brettwooldridge committed:
Update pom.xml

brettwooldridge committed:
Update .travis.yml

brettwooldridge committed:
[maven-release-plugin] prepare release HikariCP-5.0.0

brettwooldridge committed:
[maven-release-plugin] prepare for next development iteration

brettwooldridge committed:
Update README.md

brettwooldridge committed:
Update CHANGES

brettwooldridge committed:
Fix idle timeout tests, sleep too tight to idle threshold. (+1 squashed commit)
Squashed commits:
[4844f9e1] Fix idle timeout tests, sleep too tight to idle threshold. (+8 squashed commits)
Squashed commits:
[e0ec6b7f] Fix idle timeout tests, sleep too tight to idle threshold.
[c40a690d] Fix idle timeout tests, sleep too tight to idle threshold.
[7a0a1560] Synchronize idleTimeout manipulation methods
[bab9f648] Fix idle timeout tests, sleep too tight to idle threshold.
[4ee5259b] Add missing ) character in message
[e9d49fc2] Enable debug logging for tests.
[89c1ab45] Fix idle timeout tests, sleep too tight to idle threshold.
[aa06bb85] Simplify pool entry last-accessed time handling.

brettwooldridge committed:
Update dependencies.

brettwooldridge committed:
[maven-release-plugin] prepare release HikariCP-5.0.1

brettwooldridge committed:
[maven-release-plugin] prepare for next development iteration

brettwooldridge committed:
Update README.md

snyk-bot committed:
fix: pom.xml to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JAVA-ORGHIBERNATE-1041788
- https://snyk.io/vuln/SNYK-JAVA-ORGHIBERNATE-584563

dependabot[bot] committed:
Bump postgresql from 42.2.20 to 42.3.3

Bumps [postgresql](https://github.com/pgjdbc/pgjdbc) from 42.2.20 to 42.3.3.
- [Release notes](https://github.com/pgjdbc/pgjdbc/releases)
- [Changelog](https://github.com/pgjdbc/pgjdbc/blob/master/CHANGELOG.md)
- [Commits](https://github.com/pgjdbc/pgjdbc/compare/REL42.2.20...REL42.3.3)

---
updated-dependencies:
- dependency-name: org.postgresql:postgresql
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

brettwooldridge committed:
Simplify connection creation.

brettwooldridge committed:
Merge commit '724140e683e5e9e892dad9607154220a835e0132' into dev

* commit '724140e683e5e9e892dad9607154220a835e0132':
  fix: pom.xml to reduce vulnerabilities

brettwooldridge committed:
Merge commit 'b140fb3de36b3789266b5b0677065452dedbbd96' into dev

* commit 'b140fb3de36b3789266b5b0677065452dedbbd96':
  Bump postgresql from 42.2.20 to 42.3.3

brettwooldridge committed:
Update dependencies.

brettwooldridge committed:
Add log when connection not added.

spliffone committed:
chore: use valid SCM URL instad of the Git SSH URL (#1930)

#1929

Co-authored-by: Bajohr, Rayk <rayk.bajohr@siemens.com>

brettwooldridge committed:
Minor code cleanups.

brettwooldridge committed:
Add pool stats to connection acquisition timeout exception

brettwooldridge committed:
Add Librapay donations.

brettwooldridge committed:
Add Librapay donations.

kth496 committed:
Remove redundant explicit imports (#1840)

lfbayer committed:
Fix travis-ci badge urls

pjmatt committed:
Update Wix engineering writeup URL (#1984)

lfbayer committed:
Don't attempt second setNetworkTimeout call if first fails

TAregger committed:
Clarify javadoc of HikariConfig.setPoolName (#1592)

arif-banai committed:
Expanded javadoc for the default constructor of HikariConfig (#1732)

msteiger committed:
Don't log any passwords in JDBC url (#1979)

Co-authored-by: Martin Steiger <martin.steiger@ruv.de>

Wawrzyniec-Nowak committed:
Bring back automatic-module-name property (#1908)

Commit:
Add `com.zaxxer.hikari.metrics.dropwizard`, `com.zaxxer.hikari.metrics.micrometer` and `com.zaxxer.hikari.metrics.prometheus` to the OSGi exported packages. (#1989)

msteiger committed:
Don't log keystore passwords in DriverDataSource (#1993)

cuishuang committed:
fix README.md  typo (#1997)

Signed-off-by: cui fliter <imcusg@gmail.com>

Signed-off-by: cui fliter <imcusg@gmail.com>

lfbayer committed:
fix test: sleep needs to be guaranteed to be greater than aliveBypassWindowMs, not just equal

minesunny committed:
make github pipline successful (#2012)

* workflow manually from the Actions tab

workflow manually from the Actions tab

* set use jdk 11

set use jdk 11

* set default reuseForks is false

because HikariConfigTest couldn‘t get log record;

brettwooldridge committed:
Cleanup maven build and modify for Java 11+ only.

brettwooldridge committed:
Close #1986 evict connection exceptions with (SQLState HY000) error code 1105.

brettwooldridge committed:
CircleCI build config

brettwooldridge committed:
CircleCI build config

brettwooldridge committed:
circleci build badge

brettwooldridge committed:
remove travis-ci artifacts

JeffFang committed:
Remove error log for pool initialization exception (#1660)

Co-authored-by: Jeff Fang <jeff.fang@activenetwork.com>

NicolaIsotta committed:
add support for char[] in PropertyElf (#2059)

fix #2058

lfbayer committed:
fix comment typo

swalchemist committed:
refactor: fix spelling (#2046)

Co-authored-by: Bruce Ricard <bricard@vmware.com>

liamsorsby committed:
[#1950] Adds support for TRANSACTION_CURSOR_STABILITY and TRANSACTION_LAST_COMMITTED (#1952)

TRANSACTION_LAST_COMMITTED - This allows a readonly user to access a row and read the last commited data while Informix has an open lock against the data.
Unlike TRANSACTION_READ_UNCOMMITTED which would also accomplish the same but would end up allowing dirty reads. This would lead to us reading data that's potentially either rolledback or never actually commited.

TRANSACTION_CURSOR_STABILITY - Cursor stability will place a lock on a row as it's fetched which ensures that the data doesn't change while the program reads the data.
Documentation for cursor stability can be found here https://www.ibm.com/docs/en/informix-servers/12.10?topic=level-informix-cursor-stability-isolation

Documentation on Informix transaction isolation levels are here https://www.ibm.com/docs/en/informix-servers/12.10?topic=database-informix-environment-variables-informix-jdbc-driver

Co-authored-by: Liam Sorsby <liam.sorsby@skybettingandgaming.com>

cal101 committed:
[cleanup] erefactor/EclipseJdt - Evaluate without null check (#1751)

EclipseJdt cleanup 'EvaluateNullable' applied by erefactor.

For EclipseJdt see https://www.eclipse.org/eclipse/news/4.18/jdt.php
For erefactor see https://github.com/cal101/erefactor

cal101 committed:
[cleanup] erefactor/EclipseJdt - Remove redundant super() call in constructor (#1753)

EclipseJdt cleanup 'RemoveUnnecessarySuperCall' applied by erefactor.

For EclipseJdt see https://www.eclipse.org/eclipse/news/4.18/jdt.php
For erefactor see https://github.com/cal101/erefactor

cal101 committed:
[cleanup] erefactor/EclipseJdt - Simplify lambda expression and method reference syntax (#1754)

EclipseJdt cleanup 'SimplifyLambdaExpression' applied by erefactor.

For EclipseJdt see https://www.eclipse.org/eclipse/news/4.18/jdt.php
For erefactor see https://github.com/cal101/erefactor

cal101 committed:
[cleanup] erefactor/EclipseJdt - Remove trailing whitespace - All lines (#1752)

EclipseJdt cleanup 'RemoveAllTrailingWhitespace' applied by erefactor.

For EclipseJdt see https://www.eclipse.org/eclipse/news/4.18/jdt.php
For erefactor see https://github.com/cal101/erefactor

Co-authored-by: Leo Bayer <lfbayer@gmail.com>

bananayong committed:
Fix #1818 Add system property for minimum login timeout, regardless of the connectionTimeout (#1820)

alsin committed:
Don't try to cast metric registry before it is checked for null (#2076)

* Don't try to cast metric registry before it is checked for null

* Simplify the condition check

* Use previously obtained object instead of calling the same getter one more time

---------

Co-authored-by: Alexander Sinuskin <Alexander.Sinuskin.ext@deutsche-boerse.com>

lfbayer committed:
update changelog

lfbayer committed:
move oss-parent build into pom.xml and upgrade some build dependencies

lfbayer committed:
change version to 5.1.0

lfbayer committed:
[maven-release-plugin] prepare release HikariCP-5.1.0

rvilyamov committed:
Merge tag 'refs/tags/HikariCP-5.1.0' into dev

[maven-release-plugin] copy for tag HikariCP-5.1.0

# Conflicts:
#	.github/workflows/ci.yml
#	pom.xml
#	src/main/java/com/zaxxer/hikari/pool/HikariPool.java
#	src/main/java/com/zaxxer/hikari/pool/ProxyLeakTask.java

rvilyamov committed:
Merge remote-tracking branch 'origin/dev' into dev

Commit:
Merge tag 'HikariCP-5.1.0' of https://github.com/brettwooldridge/HikariCP into DB-226459

[maven-release-plugin] copy for tag HikariCP-5.1.0

Commit:
DB-226459 Update HikariCP to 5.1.0 version
DB-226459-141706 [DEV] Update HikariCP to 5.1.0 version

Changes:
1. Add folder ".settings" in .gitignore

Commit:
Merge branch 'DB-226459' of https://github.com/IKAMTeam/HikariCP into DB-226459

Commit:
DB-226459 Update HikariCP to 5.1.0 version
DB-226459-141706 [DEV] Update HikariCP to 5.1.0 version

Changes:
1. Updated the method executes "on borrow" SQL to return status of execution: "true" - OK otherwise "false".
2. Check result of "on borrow" SQL execution with other Hikari's standard checks together as the last item. In case if the "on borrow" result is false then the pool closes such conneciton and takes (or creates new) another one.

Commit:
DB-226459 Update HikariCP to 5.1.0 version
DB-226459-141706 [DEV] Update HikariCP to 5.1.0 version

Changes:
1. Removed the ".settings" folder from the repository.

asmoiseenko reviewed (commented):
Update copyright notices on files you edited, such files should have 
```
Copyright (C) 2024 OneVizion, Inc. All rights reserved.
```

Commit:
DB-226459 Update HikariCP to 5.1.0 version
DB-226459-141706 [DEV] Update HikariCP to 5.1.0 version

Changes:
1. Updated copyright's year.

skubp commented:
> Update copyright notices on files you edited, such files should have
> 
> ```
> Copyright (C) 2024 OneVizion, Inc. All rights reserved.
> ```

Done

asmoiseenko committed:
DB-226459 Update HikariCP to 5.1.0 version

DB-226459-141706 [DEV] Update HikariCP to 5.1.0 version
Updated copyright year.

asmoiseenko committed:
DB-226459 Update HikariCP to 5.1.0 version

DB-226459-141706 [DEV] Update HikariCP to 5.1.0 version
Updated copyright year.
